firmware/boron-ins-fsm/src/stateMachine.cpp

```diff
@@ -384,6 +393,18 @@ void getHeartbeat() {
         char heartbeatMessage[622] = {0};
         JSONBufferWriter writer(heartbeatMessage, sizeof(heartbeatMessage) - 1);
         writer.beginObject();
+
+        if (missedDoorEventQueue.size() > missedDoorEventTrackN) {
+            if (missedDoorEventQueue.front())
+	        missedDoorEventAmnt--; // oldest value did miss; subtract it from the current amount
+            missedDoorEventQueue.pop();
+        }
+        bool didMiss = missedDoorEventCount > 0;
+        missedDoorEventAmnt += (int)didMiss;
+        missedDoorEventQueue.push(didMiss); // enqueue whether this heartbeat did miss
+        // logs whether or not sensor is frequently missing door events
+        writer.name("doorMissedFreq").value(missedDoorEventAmnt > missedDoorEventMax);
+
         // logs number of instances of missed door events since last heartbeat
         writer.name("doorMissedMsg").value(missedDoorEventCount);
         missedDoorEventCount = 0;
```

Comment:
I'm pretty sure that door messages can come in at any time, which means they may interrupt the execution of this function. Can you think of anything weird that might happen in here with `missedDoorEventCount`?

Reply:
Good job identifying what I was pointing out. What you did to store an `instant` version of the variable and immediately then setting it to 0 is what I was thinking too. Admittedly, this doesn't eliminate the issue, it just makes it less likely to occur. To completely eliminate the issue, you would have needed to start looking into threadsafe data structures. What you've done is good enough for our purposes.